src/hotspot/share/code/nmethod.cpp

```diff
@@ -2127,9 +2149,13 @@ void nmethod::purge(bool unregister_nmethod) {
   }
   delete[] _compiled_ic_data;
 
-  if (_immutable_data != data_end()) {
+  if (_immutable_data != blob_end()) {
```

Comment:
Is this just a name change, or a semantic change?

Reply:
In several places _immutable_data is set to data_end() "valid not null address" address which was actually the end of the code blob. With this change I remove the _data part of the code blob as well as the data_begin() and data_end() functions. I think blob_end() is a good replacement for data_end() for empty _immutable_data cases.